clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10151,6 +10151,9 @@ def err_sycl_non_std_layout_type : Error<
   "kernel parameter has non-standard layout class/struct type">;
 def err_conflicting_sycl_kernel_attributes : Error<
   "conflicting attributes applied to a SYCL kernel">;
+def err_sycl_function_attributes_conflict : Error<
```

Comment:
Please consider following code:
`[[attr(1)]] void boo();
[[attr(2)]] void goo();

void foo() {
  boo();
  goo();
}`

For previously added kernel attributes we allowed a left-side spelling (aka attribute-on-a-function, not attribute-on-a-function-type) with propagation of this attribute from the callee to a caller. It could result in a conflict during this propagation if the callee function had the same attribute applied, but with different arguments. `err_conflicting_sycl_kernel_attributes` diagnostics was previously added to handle these cases.

Here I add another diagnostic (`err_sycl_function_attributes_conflict`) for another purpose:
```
kernel<class test_kernel6>(
 []() [[cl::reqd_work_group_size(16, 2, 2),
         intelfpga::max_work_group_size(2, 2, 2)]] {});
```

here we explicitly add two different attributes to the same lambda. `max_work_group_size` arguments X, Y, Z shall limit possible values of `reqd_work_group_size` arguments, but in this case X argument of `reqd_work_group_size `is higher then X argument of `max_work_group_size` and so the diagnostics is emitted.

Reply:
Thanks for outlining this, as I better understand the concept now. My main suggestion was to simply rename the error though :)

Reply:
@AGindinson ah, sorry, I got confused by the fact that there is a warning called: `err_conflicting_sycl_kernel_attributes` so I mixed up your suggestion with what I saw in DiagnosticSemaKinds.td and decided, that you want to merge these diagnostics into one :D